sdk/storage/azdatalake/directory/client.go

```diff
@@ -25,11 +25,41 @@ type ClientOptions base.ClientOptions
 // Client represents a URL to the Azure Datalake Storage service.
 type Client base.CompositeClient[generated.PathClient, generated.PathClient, blob.Client]
 
-//TODO: NewClient()
+// NewClient creates an instance of Client with the specified values.
+//   - directoryURL - the URL of the directory e.g. https://<account>.dfs.core.windows.net/fs/dir
+//   - cred - an Azure AD credential, typically obtained via the azidentity module
+//   - options - client options; pass nil to accept the default values
+func NewClient(directoryURL string, cred azcore.TokenCredential, options *ClientOptions) (*Client, error) {
+	blobURL := strings.Replace(directoryURL, ".dfs.", ".blob.", 1)
+	directoryURL = strings.Replace(directoryURL, ".blob.", ".dfs.", 1)
+
+	authPolicy := shared.NewStorageChallengePolicy(cred)
```

Comment:
StorageChallengePolicy shouldnt be needed since this is explicitly for managed disks

Reply:
you're right, I'll change in my next pr